primitives/core/src/ecdsa.rs

```diff
@@ -648,6 +648,20 @@ mod test {
 		assert_eq!(cmp, public);
 	}
 
+	#[test]
+	fn ss58check_custom_format_works() {
+		use crate::crypto::Ss58AddressFormat;
+		let default_format = Ss58AddressFormat::default();
+		set_default_ss58_version(Ss58AddressFormat::try_from("200").expect(""));
+		// custom
+		let addr = "2X64kMNEWAW5KLZMSKcGKEc96MyuaRsRUku7vomuYxKgqjVCRj";
+		let _ = Public::from_ss58check(&addr).unwrap();
```

Comment:
also maybe replace `unwrap` with `expect` with some error description?

Reply:
ok I fix it